cmd/syncthing/main.go

```diff
@@ -957,16 +971,81 @@ func loadConfigAtStartup() *config.Wrapper {
 		l.Fatalln("Failed to load config:", err)
 	}
 
-	if cfg.RawCopy().OriginalVersion != config.CurrentVersion {
+	configVersion := cfg.RawCopy().OriginalVersion
+
+	if configVersion != config.CurrentVersion {
+		// Check config version
+		if cfg.RawCopy().OriginalVersion > config.CurrentVersion {
+			// Config version is not compatible with installed version
+			if allowDowngrade {
+				l.Infof("Current Configuration Version: %d, Loading Configuration Version: %d (Configuration Version Mismatch)", configVersion, config.CurrentVersion)
+				l.Infof("Configuration downgrading...")
+
+				fd, err := os.Open(cfgFile)
+				if err != nil {
+					l.Fatalln("Config file not opened")
+				}
+				defer fd.Close()
+
+				// Read XML and set defaults (Check-1)
+				// In most cases everything should correct on here  except Version.
+				cfgXML, err := config.ReadXML(fd, myID)
+				if err != nil {
+					l.Fatalln("Config not read")
+				}
```

Comment:
What's the point of reading it again? `cfg, err := config.Load` further up did exactly the same thing. And below you are actually passing `cfg` to `archiveAndSaveConfig`. As far as I understand you can just replace the only occurrence of `cfgXML` with `cfg.RawCopy()`.